File: solutions/1652. Defuse the Bomb/1652.py
# STATS:
# Runtime = 42ms
# memory = 16.54mb
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:        
        n = len(code)
        arr = code*2
        decrypted_code = []

        for i in range(n):
            if k > 0:
                decrypted_code.append(sum(arr[i+1:i+k+1]))
            elif k < 0:
                decrypted_code.append(sum(arr[i+n-(-k):i+n]))
            else:
                decrypted_code = [0]*len(code)
        return decrypted_code

File: solutions/1652. Defuse the Bomb/1652.rb
# @param {Integer[]} code
# @param {Integer} k
# @return {Integer[]}
def decrypt(code, k)
    return [0]*code.size if k == 0
    
    decrypted_code=[]
    n=code.size
    w_start = k > 0 ? 1 : n+k
    w_end = k > 0 ? k : n-1
    sum=0
    (w_start..w_end).each { |i| sum += code[i%n]} #initial sum
    decrypted_code << sum

    (1...n).each do |i|
        sum += code[(w_end+1)%n]
        sum -= code[w_start%n]
        decrypted_code << sum

        w_start += 1
        w_end += 1
    end
    decrypted_code
end
